src/mds/MDCache.cc

```diff
@@ -5925,7 +5925,7 @@ bool MDCache::open_undef_inodes_dirfrags()
   MDSGatherBuilder gather(g_ceph_context,
       new MDSInternalContextWrapper(mds,
 	new LambdaContext([this](int r) {
-	    if (rejoin_gather.empty())
+	    if (rejoin_gather.empty() && rejoin_ack_gather.count(mds->get_nodeid()))
```

Comment:
open_undef_inodes_dirfrags() is only called by rejoin_gather_finish(). I don't think it's the root cause.